<?xml version="1.0" encoding="UTF-8"?>
<definitions xmlns="http://www.omg.org/spec/BPMN/20100524/MODEL"
  xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema"
  xmlns:activiti="http://activiti.org/bpmn" xmlns:bpmndi="http://www.omg.org/spec/BPMN/20100524/DI"
  xmlns:omgdc="http://www.omg.org/spec/DD/20100524/DC"
  xmlns:omgdi="http://www.omg.org/spec/DD/20100524/DI"
  typeLanguage="http://www.w3.org/2001/XMLSchema" expressionLanguage="http://www.w3.org/1999/XPath"
  targetNamespace="http://www.activiti.org/test">
  <process id="AMA Map Open Toolkit DITA To PDF" name="AMA Map Open Toolkit DITA To PDF" isExecutable="true">
    <startEvent id="startWorkflow" name="Start"/>
    
    <sequenceFlow id="flowToEnsureMapForProductTask" sourceRef="startWorkflow" targetRef="ensureMapForProductTask"/>
    
    <serviceTask id="ensureMapForProductTask" name="Ensure Map" activiti:class="org.ama_assn.rsuite.workflow.actions.leaving.EnsureMapActionHandler">
      <activiti:field name="forceNewMapFromWorkflow" stringValue="true"/>
      <activiti:field name="xsltUriFromWorkflow" stringValue="rsuite:/res/plugin/ama-main/xslt/ca2map/product_ca2map_shell.xsl"/>
    </serviceTask>
    
    
    <sequenceFlow id="flowToEnsureMapDecisionError" sourceRef="ensureMapForProductTask" targetRef="ensureMapDecisionErrorGateway"></sequenceFlow>
    
    <exclusiveGateway id="ensureMapDecisionErrorGateway" name="EnsureMap Error?"></exclusiveGateway>
    <sequenceFlow id="flowToGetMoDetails" sourceRef="ensureMapDecisionErrorGateway" targetRef="getMoDetailsTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "false"}]]></conditionExpression>
    </sequenceFlow>
    
    <sequenceFlow id="flowToEnsureMapFailure" sourceRef="ensureMapDecisionErrorGateway" targetRef="ensureMapDecisionErrorTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "true"}]]></conditionExpression>
    </sequenceFlow>
    <userTask id="ensureMapDecisionErrorTask" name="Map failed to load">
      <documentation>${taskFullDescription}</documentation>
    </userTask>
    <sequenceFlow id="flowEnsureMapDecisionErrorTaskToEnd" sourceRef="ensureMapDecisionErrorTask" targetRef="endWorkflow"></sequenceFlow>
    
    <serviceTask id="getMoDetailsTask" name="Get MO Details" activiti:class="com.reallysi.rsuite.system.workflow.actions.SetVariablesFromMoDetails"></serviceTask>
    
    <sequenceFlow id="flowToMapToOutput" sourceRef="getMoDetailsTask" targetRef="runOTGenerationTask"/>
    
    <serviceTask id="runOTGenerationTask" name="Run Open Toolkit generation" activiti:class="org.ama_assn.rsuite.workflow.actions.leaving.rsuite5.ProjectDitaOtPdfTaskRunningActionHandler">
      <activiti:field name="buildPropertiesFromWorkflow" stringValue="showTrackChanges=${showTrackChanges}|showComments=${showComments}" />
      <activiti:field name="transtypeFromWorkflow" expression="${transtype}"/>
      <activiti:field name="outputPathFromWorkflow" expression="${rsuiteDitaXmlExportDir}/${rsuiteUserId}/pdf"/>
    </serviceTask>
    
    <sequenceFlow id="flowToDitaOTDecision" sourceRef="runOTGenerationTask" targetRef="ditaOTError"></sequenceFlow>
    
    <exclusiveGateway id="ditaOTError" name="DITAOT Error?"></exclusiveGateway>
    <sequenceFlow id="flowToPrepareResult" sourceRef="ditaOTError" targetRef="prepareResultTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "false"}]]></conditionExpression>
    </sequenceFlow>
    <serviceTask id="prepareResultTask" name="Prepare Result" activiti:class="com.reallysi.rsuite.workflow.actions.PrepareResultToWorkflowFileObjectActionHandler"></serviceTask>
    <sequenceFlow id="flowToLoadResult" sourceRef="prepareResultTask" targetRef="loadResultsTask"></sequenceFlow>
    <serviceTask id="loadResultsTask" name="Load Result" activiti:class="org.ama_assn.rsuite.workflow.actions.leaving.rsuite5.ProjectLoadNonXmlByAliasHandler">
      <activiti:field name="commitMessageFromWorkflow" expression="Generated by ${rsuiteUserId} via workflow"/>
      <activiti:field name="parentMoIdFromWorkflow" expression="${idToLoadInto}"/>
      <activiti:field name="fileToLoadFromWorkflow" stringValue=""/>
    </serviceTask>
    
    <sequenceFlow id="flowToAttachResult" sourceRef="loadResultsTask" targetRef="attachResultsTask"></sequenceFlow>
    <serviceTask id="attachResultsTask" name="Attach Result to Workflow" activiti:class="com.reallysi.rsuite.workflow.actions.AttachMoToWorkflowAction"></serviceTask>
    
    
    <sequenceFlow id="flowToGenerateReport" sourceRef="attachResultsTask" targetRef="generateReport"></sequenceFlow>
    <serviceTask id="generateReport" name="Generate Report" activiti:class="com.reallysi.rsuite.workflow.actions.GenerateReportAction"></serviceTask>
    <sequenceFlow id="flowToLoadErrorDecision" sourceRef="generateReport" targetRef="loadErrorGateway"></sequenceFlow>
    <exclusiveGateway id="loadErrorGateway" name="Load Error?"></exclusiveGateway>
    <sequenceFlow id="flowToViewReport" sourceRef="loadErrorGateway" targetRef="viewReportTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "false"}]]></conditionExpression>
    </sequenceFlow>
    <userTask id="viewReportTask" name="View Report" activiti:assignee="${rsuiteUserId}" activiti:candidateGroups="RSuiteAdministrator">
      <documentation>${taskFullDescription}</documentation>
    </userTask>
    <sequenceFlow id="flowFromViewReportToEnd" sourceRef="viewReportTask" targetRef="endWorkflow"></sequenceFlow>
    <sequenceFlow id="flowToLoadFailure" sourceRef="loadErrorGateway" targetRef="loadFailureTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "true"}]]></conditionExpression>
    </sequenceFlow>
    <userTask id="loadFailureTask" name="Load Failure">
      <documentation>${taskFullDescription}</documentation>
    </userTask>
    <sequenceFlow id="flowLoadFailureToEnd" sourceRef="loadFailureTask" targetRef="endWorkflow"></sequenceFlow>
    <sequenceFlow id="flowToGenFailSetUp" sourceRef="ditaOTError" targetRef="outputGenFailSetUpTask">
      <conditionExpression xsi:type="tFormalExpression"><![CDATA[${EXCEPTION_OCCUR == "true"}]]></conditionExpression>
    </sequenceFlow>
    <serviceTask id="outputGenFailSetUpTask" name="Output Generation Fail Set Up" activiti:class="com.reallysi.rsuite.workflow.actions.GenerationFailSetUpAction"></serviceTask>
    <sequenceFlow id="flowToAssignTaskToVar" sourceRef="outputGenFailSetUpTask" targetRef="generationFailureTask"></sequenceFlow>
    <userTask id="generationFailureTask" name="Output Generation Failure" activiti:assignee="${rsuiteUserId}" activiti:candidateGroups="RSuiteUser">
      <documentation>${taskFullDescription}</documentation>
    </userTask>
    <sequenceFlow id="flowToEndWorkflow" sourceRef="generationFailureTask" targetRef="endWorkflow"></sequenceFlow>
    <endEvent id="endWorkflow" name="End Workflow"></endEvent>
    <dataObject id="EXCEPTION_OCCUR" name="EXCEPTION_OCCUR" itemSubjectRef="xsd:string">
      <extensionElements>
        <activiti:value>""</activiti:value>
      </extensionElements>
    </dataObject>
    <dataObject id="rsuiteUserId" name="rsuiteUserId" itemSubjectRef="xsd:string"></dataObject>
  </process>
  
</definitions>
